pkgs/dart_mcp/lib/src/api/api.dart

```diff
@@ -293,31 +301,41 @@ extension type EmbeddedResource.fromMap(Map<String, Object?> _value)
   String? get mimeType => _value['mimeType'] as String?;
 }
 
-// /**
-//  * Base for objects that include optional annotations for the client. The
-//  * client can use annotations to inform how objects are used or displayed
-//  */
-// export interface Annotated {
-//   annotations?: {
-//     /**
-//      * Describes who the intended customer of this object or data is.
-//      *
-//      * It can include multiple entries to indicate content useful for
-//      * multiple audiences (e.g., `["user", "assistant"]`).
-//      */
-//     audience?: Role[];
-
-//     /**
-//      * Describes how important this data is for operating the server.
-//      *
-//      * A value of 1 means "most important," and indicates that the data is
-//      * effectively required, while 0 means "least important," and indicates
-//      * that the data is entirely optional.
-//      *
-//      * @TJS-type number
-//      * @minimum 0
-//      * @maximum 1
-//      */
-//     priority?: number;
-//   }
-// }
+/// Base type for objects that include optional annotations for the client.
+///
+/// The client can use annotations to inform how objects are used or displayed.
+extension type Annotated._fromMap(Map<String, Object?> _value) {
+  /// Annotations for this object.
+  Annotations? get annotations => _value['annotations'] as Annotations?;
+}
+
+/// The annotations for an [Annotated] object.
+extension type Annotations.fromMap(Map<String, Object?> _value) {
+  factory Annotations({List<Role>? audience, double? priority}) =>
+      Annotations.fromMap({
```

Comment:
should we assert that priority is between 0 and 1?

Reply:
done, this won't happen when casting from a Map, but the main use case is when going through this "constructor" anyways.